src/botPage/view/View.js

```diff
@@ -467,8 +468,19 @@ export default class View {
         });
     }
     stop() {
+        try {
+            const { contractId } = this.blockly.interpreter.bot.tradeEngine;
+            if (contractId) {
+                api.subscribeToOpenContract(contractId);
+            }
+            api.events.on('proposal_open_contract', r => {
```

Comment:
This should be outside of the stop function, otherwise will be called per stop call and will result in multiple broadcast for each open contract.